### ISSUE TITLE
Is the workload spec in the Component spec filled by developers or operators?

### ISSUE THREAD

**Issue by doudoutinghua** [oam-dev/spec#380]
**Is the content in the red box filled by developers or operators?**

![image](https://user-images.githubusercontent.com/43887164/87539083-e3c3a080-c6cf-11ea-9cc4-b0e428b838f5.png)

![image](https://user-images.githubusercontent.com/43887164/87539107-e9b98180-c6cf-11ea-9a1c-88e3dc68cb0b.png)

![image](https://user-images.githubusercontent.com/43887164/87539137-f8079d80-c6cf-11ea-8de7-d6e61633d086.png)

**Comment by wonderflow** [comment]
It should be filled by developers

**Comment by vturecek** [comment]
Yes this is where developers fill in the detail of the container they want to run for a container workload. More generally, for any workload type, developers are responsible for producing a component file. 

If that answers your question, @doudoutinghua, I will close this issue out.

**Comment by babin93** [comment]
Where is the workload schema defined ? Is it directly in the workload ? I can't easily find a complete example of interactions between a workload and a component.

**Comment by resouer** [comment]
@babin93 For ContainerizedWorkload which is part of OAM spec, its schema is defined in this doc: https://github.com/oam-dev/spec/blob/master/core/workloads/containerized_workload/containerized_workload.md

For any other workload, its schema is deferred to underlying runtime. OAM will use `definitionRef` in https://github.com/oam-dev/spec/blob/master/3.workload.md#3-workload-definition to track the schema. In Kubernetes for example, `definitionRef ` will point to a specific Kubernetes API resource (built-in or CRD).

That is saying, for any application platform build with OAM model, it will rely on `WorkloadDefinition` to let the developers know how to use certain workload. Note that`WorkloadDefinition` is a extensible object so platform builders are encouraged to reference doc link or API spec doc link in that object to help developers understand the workload better.

Hope this answered your question and feel free to raise any discussion around the spec.
